pylint/lint/report_functions.py

```diff
@@ -54,14 +54,12 @@ def report_messages_by_module_stats(
     if len(module_stats) == 1:
         # don't print this report when we are analysing a single module
         raise exceptions.EmptyReportError()
-    by_mod: DefaultDict[str, Dict[str, Union[int, float]]] = collections.defaultdict(
-        dict
-    )
+    by_mod: DefaultDict = collections.defaultdict(dict)
     for m_type in ("fatal", "error", "warning", "refactor", "convention"):
-        total: int = stats[m_type]  # type: ignore
-        for module in module_stats.keys():
-            mod_total = module_stats[module][m_type]
-            percent = 0 if total == 0 else float((mod_total) * 100) / total
+        total = stats[m_type]
+        for module in stats["by_module"].keys():
```

Comment:
I'm fine with this, although we should then remove L53.

Reply:
After latest commit I think we should use L53 so we can remove some of the `type: ignore` on these lines.